design/implicit-boot-mode.md

```diff
@@ -0,0 +1,197 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# implicit-boot-mode
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+## Summary
+
+The required boot mode for a host depends on several factors,
+including support in the hardware, whether IPv6 is being used, and
+whether virtual media is being used. We do not need to support all
+combinations of those features with all boot modes. Asking the user to
+give us the boot mode explicitly introduces an opportunity for them to
+specify an incorrect value. We have enough information about the host
+already to determine the correct boot mode to use. Therefore we should
+not need to extend the BareMetalHost API in order to support multiple
+boot modes.
+
+## Motivation
+
+When booting using virtual media, Ironic builds a custom ISO for each
```

Comment:
Isn't there the possibility of building a hybrid ISO that can work in both modes? This will save us complexity

Reply:
This is technically feasible, but needs to be coded in ironic.

May be we could build hybrid ISO only if ironic is not certain what boot mode is in effect.

Reply:
That's certainly another possible approach. Here I am trying to limit the scope of work. I think we can make this work with 1 patch in the baremetal-operator repository.